### PR TITLE
#9792 Migrate Anchor to new lifecycle methods

### PR DESCRIPTION
**Pull request by natanielcz**
Part of #9792

### CODE REVIEW

File: components/anchor/AnchorLink.tsx [hunk 1]
```diff
@@ -77,4 +91,27 @@ export default class AnchorLink extends React.Component<AnchorLinkProps, any> {
...
+  private handleClick = (e: React.MouseEvent<HTMLElement>) => {
+    const {scrollTo, onClick} = this.context.antAnchor;
+    const {href, title} = this.props;
+    if (onClick) {
+      onClick(e, {title, href});
+    }
+    scrollTo(href);
+  };
+
+  private registerLink(): void {
```
Comment (afc163) [inline components/anchor/AnchorLink.tsx]: No need to add private or public under this issue.

Comment (natanielcz) [reply components/anchor/AnchorLink.tsx]: This is a small fix. I thought that It would be nice because it doesn't take me a lot of time.

Comment (yesmeck) [reply components/anchor/AnchorLink.tsx]: Please follow the exits code style.

File: components/anchor/Anchor.tsx [hunk 2]
```diff
@@ -233,11 +233,11 @@ export default class Anchor extends React.Component<AnchorProps, AnchorState> {
     if (linkNode) {
       this.inkNode.style.top = `${(linkNode as any).offsetTop + linkNode.clientHeight / 2 - 4.5}px`;
     }
-  }
+  };
 
   saveInkNode = (node: HTMLSpanElement) => {
     this.inkNode = node;
-  }
+  };
```
Comment (yesmeck) [inline components/anchor/Anchor.tsx]: Same here, these changes are not related to the subject of this PR.

**Comment by natanielcz** [comment]
I removed optional changes.

**Comment by damiangreen** [comment]
This LGTM. It appears the submitted has addressed the issues
